### PR TITLE
refactor(write): normalize deletes field

### PR DESCRIPTION
**Pull request by jpadilla**
## Description

Regenerate based on latest changes in https://github.com/openfga/api/pull/98

## References

https://github.com/openfga/api/pull/116
https://github.com/openfga/api/pull/98
https://github.com/openfga/sdk-generator/pull/236

## Review Checklist
- [ ] I have clicked on ["allow edits by maintainers"](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork).
- [ ] I have added documentation for new/changed functionality in this PR or in a PR to [openfga.dev](https://github.com/openfga/openfga.dev) [Provide a link to any relevant PRs in the references section above]
- [ ] The correct base branch is being used, if not `main`
- [ ] I have added tests to validate that the change in functionality is working as expected

### CODE REVIEW

File: client/client.go [hunk 1]
```diff
@@ -1157,7 +1158,7 @@ type SdkClientWriteRequestInterface interface {
 
 type ClientWriteRequest struct {
 	Writes  []ClientWriteRequestTupleKey
-	Deletes []ClientWriteRequestTupleKey
+	Deletes []ClientTupleKeyWithoutCondition
```
Comment (adriantam) [inline client/client.go:1160]: Do we need to update the README examples (i.e., https://github.com/openfga/go-sdk/blob/c3ee17eb6a0558f99c134fa403ad2d6af4e48144/README.md?plain=1#L464)?

Comment (jpadilla) [reply client/client.go:1160]: oh yeah thanks, seems like they were already slightly out of date.

File: go.mod [hunk 2]
```diff
@@ -4,5 +4,5 @@ go 1.21.1
 
 require (
 	github.com/jarcoal/httpmock v1.3.1
-	golang.org/x/sync v0.5.0
```
Comment (jon-whit) [inline go.mod:7]: Was this intentional?

Comment (jpadilla) [reply go.mod:7]: hmm i don't think so, let me look at this.

**Pull request merged**